### ISSUE TITLE
question about  perception test dataset

### ISSUE THREAD

**Issue by yuhanying** [JacobChalk/TIM#43]
Thank you for your great work on this project. I have a question regarding the process of recreating the perception test dataset.

I noticed that when using the `--train` flag, the **validation set** is employed to monitor model performance and to save the best-performing checkpoint(https://github.com/JacobChalk/TIM/blob/c7cb2935fb6736db3e4328e749ad25a2433a1654/recognition/scripts/train.py#L59-L136). However, when using the `--validate` flag, it appears that the same validation set is used to evaluate the model.

Given this setup, wouldn't the evaluation results be based on data that the model has already been exposed to during training? If so, could this potentially compromise the validity or objectivity of the evaluation?

I would greatly appreciate any clarification on this matter.
Thank you for your time and support.

**Comment by JacobChalk** [comment]
Hi,

The validation set is a withheld subset of the training data. In our case, we use exclusively use the train set to train the model and use the validation set only for evaluation. Indeed, it is common that people combine train+validation when submitting to challenges on test sets, meaning eventually the model will see all the data, but this is also usually after hyper parameter tuning on the validation set (where at that point, it is unseen). 

If you were to train on all the data and then validate on the same validation, then yes the evaluation would be compromised. However, in our case we do not have access to any evaluation sets directly during training I.e. we only ever train with the train set and the validation set is never seen during training, retaining the validity and correctness of evaluation on an unseen subset of the total data!
